zkevm-circuits/src/copy_circuit.rs

```diff
@@ -328,8 +333,8 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                 meta.query_advice(rw_counter, Rotation::cur()),
                 not::expr(meta.query_selector(q_step)),
                 RwTableTag::Memory.expr(),
-                meta.query_advice(id, Rotation::cur()), // call_id
```

Comment:
For `TxLog`, `Memory`, `Tx calldata` it seems we need to constraint hi part to be exactly 0. Right now only constrain `row.id[cur].hi == row.id[cur+2].hi` but hi can be anything and escape lookup

Reply:
yes 👍 if evm circuit use it correctly, but consider from design perspective, since copy circuit responsibility is assure copy table correctness,  it somehow shouldn't assume the upper layer who lookup copy table always have better constraints. (Giving `hi` part is allocated as a cell by mistake instead of constant `0.expr(), then this case hi part can be anything)

Reply:
I added the constraint for clarity and confidence. We can optimize them in the future. Fixed in [6eb66d5](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1433/commits/6eb66d5aefe33cc75a78180a33029e2887533eb2)